# Data Pre-processing

**Import datasets into python**

In [ ]:
import pandas as pd
match_df = pd.read_csv(r"PATH TO AGG DATASET")
match_df

date  game_size  \
0         2017-11-26T20:59:40+0000         37   
1         2017-11-26T20:59:40+0000         37   
2         2017-11-26T20:59:40+0000         37   
3         2017-11-26T20:59:40+0000         37   
4         2017-11-26T20:59:40+0000         37   
...                            ...        ...   
13849282  2018-01-04T04:12:58+0000         46   
13849283  2018-01-04T04:12:58+0000         46   
13849284  2018-01-04T04:12:58+0000         46   
13849285  2018-01-04T04:12:58+0000         46   
13849286  2018-01-04T04:12:58+0000         46   

                                                   match_id match_mode  \
0         2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
1         2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
2         2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
3         2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
4         2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
...                                                     ...        ...   
13849282  2U4GBNA0YmlWc_jEcSfkfTtWimxqUoNXDijilJ0EXsIlud...        tpp   
13849283  2U4GBNA0YmlWc_jEcSfkfTtWimxqUoNXDijilJ0EXsIlud...        tpp   
13849284  2U4GBNA0YmlWc_jEcSfkfTtWimxqUoNXDijilJ0EXsIlud...        tpp   
13849285  2U4GBNA0YmlWc_jEcSfkfTtWimxqUoNXDijilJ0EXsIlud...        tpp   
13849286  2U4GBNA0YmlWc_jEcSfkfTtWimxqUoNXDijilJ0EXsIlud...        tpp   

          party_size  player_assists  player_dbno  player_dist_ride  \
0                  2               0            1       2870.724000   
1                  2               0            1       2938.407230   
2                  2               0            0          0.000000   
3                  2               0            0          0.000000   
4                  2               0            0       2619.077390   
...              ...             ...          ...               ...   
13849282           2               1            1          0.000000   
13849283           2               0            0        592.562439   
13849284           2               0            1        900.445100   
13849285           2               0            1          0.000000   
13849286           2               0            1          0.000000   

          player_dist_walk  player_dmg  player_kills     player_name  \
0              1784.847780         117             1        SnuffIes   
1              1756.079710         127             1          Ozon3r   
2               224.157562          67             0          bovize   
3                92.935150           0             0         sbahn87   
4              2510.447000         175             2       GeminiZZZ   
...                    ...         ...           ...             ...   
13849282        107.555237         187             0        YoungPee   
13849283        571.859300          51             0  KnightShade_13   
13849284        454.024000          73             1     XxXRynumXxX   
13849285         36.453487          99             1     Unknown0311   
13849286        304.118866         200             2      DoctorHrmm   

          player_survive_time  team_id  team_placement  
0                    1106.320        4              18  
1                    1106.315        4              18  
2                     235.558        5              33  
3                     197.553        5              33  
4                    1537.495       14              11  
...                       ...      ...             ...  
13849282              174.608       20              39  
13849283              355.901       31              29  
13849284              355.896       31              29  
13849285              128.873       35              32  
13849286              299.494       35              32  

[13849287 rows x 15 columns]

In [ ]:
kill_df = pd.read_csv(r"PATH TO KILL DATASET")
kill_df

killed_by       killer_name  killer_placement  killer_position_x  \
0              Grenade   KrazyPortuguese               5.0          657725.10   
1               SCAR-L  nide2Bxiaojiejie              31.0           93091.37   
2                 S686          Ascholes              43.0          366921.40   
3         Down and Out        Weirdo7777               9.0          472014.20   
4                 M416         Solayuki1               9.0          473357.80   
...                ...               ...               ...                ...   
13426343      Bluezone               NaN               NaN                NaN   
13426344  Down and Out         Subastral               9.0          298162.80   
13426345        SCAR-L            TingFg              20.0          447251.20   
13426346        SCAR-L         OldBlack6               3.0          152108.80   
13426347  Down and Out         liuyang23              19.0          446082.90   

          killer_position_y      map  \
0                  146275.2  MIRAMAR   
1                  722236.4  MIRAMAR   
2                  421623.9  MIRAMAR   
3                  313274.8  MIRAMAR   
4                  318340.5  MIRAMAR   
...                     ...      ...   
13426343                NaN  ERANGEL   
13426344           501961.7  ERANGEL   
13426345           628701.7  ERANGEL   
13426346           315162.6  ERANGEL   
13426347           627984.4  ERANGEL   

                                                   match_id  time  \
0         2U4GBNA0YmnLSqvEycnTjo-KT000vfUnhSA2vfVhVPe1QB...   823   
1         2U4GBNA0YmnLSqvEycnTjo-KT000vfUnhSA2vfVhVPe1QB...   194   
2         2U4GBNA0YmnLSqvEycnTjo-KT000vfUnhSA2vfVhVPe1QB...   103   
3         2U4GBNA0YmnLSqvEycnTjo-KT000vfUnhSA2vfVhVPe1QB...  1018   
4         2U4GBNA0YmnLSqvEycnTjo-KT000vfUnhSA2vfVhVPe1QB...  1018   
...                                                     ...   ...   
13426343  2U4GBNA0YmkTZQvSzreuDJjVgxqK1nIRqCKc5yGZ5INIVZ...  1519   
13426344  2U4GBNA0YmkTZQvSzreuDJjVgxqK1nIRqCKc5yGZ5INIVZ...  1086   
13426345  2U4GBNA0YmkTZQvSzreuDJjVgxqK1nIRqCKc5yGZ5INIVZ...   327   
13426346  2U4GBNA0YmkTZQvSzreuDJjVgxqK1nIRqCKc5yGZ5INIVZ...   624   
13426347  2U4GBNA0YmkTZQvSzreuDJjVgxqK1nIRqCKc5yGZ5INIVZ...   344   

              victim_name  victim_placement  victim_position_x  \
0         KrazyPortuguese               5.0          657725.10   
1             X3evolution              33.0           92238.68   
2                 CtrlZee              46.0          367304.50   
3               BlackDpre              13.0          476645.90   
4                   Vjolt              13.0          473588.50   
...                   ...               ...                ...   
13426343       Carmen_Amy              28.0          165267.30   
13426344       JimmyChiu_              28.0          299201.80   
13426345         LYB_LaoJ              19.0          447025.30   
13426346    jaydenhuang07              11.0          152887.40   
13426347           TingFg              20.0          447621.90   

          victim_position_y  
0                  146275.2  
1                  723375.1  
2                  421216.1  
3                  316758.4  
4                  318418.8  
...                     ...  
13426343           601221.8  
13426344           504790.9  
13426345           628270.8  
13426346           315209.8  
13426347           628611.0  

[13426348 rows x 12 columns]

**1.Data Reduction (Sampling - selecting a random subset of the original data for analysis):** 
    <br/>
>Delete rows while keeping only the first 10000 rows for both datasets to use

In [25]:
match_df.drop(match_df.index[10000:], inplace=True)
match_df

date  game_size  \
0     2017-11-26T20:59:40+0000         37   
1     2017-11-26T20:59:40+0000         37   
2     2017-11-26T20:59:40+0000         37   
3     2017-11-26T20:59:40+0000         37   
4     2017-11-26T20:59:40+0000         37   
...                        ...        ...   
9995  2017-11-20T01:00:32+0000         29   
9996  2017-11-20T01:00:32+0000         29   
9997  2017-11-20T01:00:32+0000         29   
9998  2017-11-20T01:00:32+0000         29   
9999  2017-11-20T01:00:32+0000         29   

                                               match_id match_mode  \
0     2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
1     2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
2     2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
3     2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
4     2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
...                                                 ...        ...   
9995  2U4GBNA0Ymlm4JHKbbmQ9x9_rZbdELcOlVMjnXNbpV6MTm...        tpp   
9996  2U4GBNA0Ymlm4JHKbbmQ9x9_rZbdELcOlVMjnXNbpV6MTm...        tpp   
9997  2U4GBNA0Ymlm4JHKbbmQ9x9_rZbdELcOlVMjnXNbpV6MTm...        tpp   
9998  2U4GBNA0Ymlm4JHKbbmQ9x9_rZbdELcOlVMjnXNbpV6MTm...        tpp   
9999  2U4GBNA0Ymlm4JHKbbmQ9x9_rZbdELcOlVMjnXNbpV6MTm...        tpp   

      party_size  player_assists  player_dbno  player_dist_ride  \
0              2               0            1        2870.72400   
1              2               0            1        2938.40723   
2              2               0            0           0.00000   
3              2               0            0           0.00000   
4              2               0            0        2619.07739   
...          ...             ...          ...               ...   
9995           4               0            2        5947.79000   
9996           4               0            0        4421.93262   
9997           4               0            2           0.00000   
9998           4               0            1           0.00000   
9999           4               1            0           0.00000   

      player_dist_walk  player_dmg  player_kills    player_name  \
0          1784.847780         117             1       SnuffIes   
1          1756.079710         127             1         Ozon3r   
2           224.157562          67             0         bovize   
3            92.935150           0             0        sbahn87   
4          2510.447000         175             2      GeminiZZZ   
...                ...         ...           ...            ...   
9995       1629.568000          87             2       2thikk4u   
9996       3212.797850         137             2   peckerpecker   
9997        873.056300         193             0      Bawngfist   
9998        744.918152          87             1  Coots_McGoots   
9999        670.910339          12             0          sbwwt   

      player_survive_time  team_id  team_placement  
0                1106.320        4              18  
1                1106.315        4              18  
2                 235.558        5              33  
3                 197.553        5              33  
4                1537.495       14              11  
...                   ...      ...             ...  
9995             1331.496        6               7  
9996             1161.466        6               7  
9997              361.651        6               7  
9998              497.617        7              14  
9999              495.147        7              14  

[10000 rows x 15 columns]

In [26]:
kill_df.drop(kill_df.index[10000:], inplace=True)
kill_df

killed_by       killer_name  killer_placement  killer_position_x  \
0          Grenade   KrazyPortuguese               5.0          657725.10   
1           SCAR-L  nide2Bxiaojiejie              31.0           93091.37   
2             S686          Ascholes              43.0          366921.40   
3     Down and Out        Weirdo7777               9.0          472014.20   
4             M416         Solayuki1               9.0          473357.80   
...            ...               ...               ...                ...   
9995           SKS   dragonfruitbamf               2.0          536215.70   
9996          M416            lddoos               3.0          550700.40   
9997  Down and Out             Nan_P               3.0          507691.30   
9998       Grenade          brentech               7.0          503235.20   
9999        SCAR-L              SwwH              13.0          447508.20   

      killer_position_y      map  \
0              146275.2  MIRAMAR   
1              722236.4  MIRAMAR   
2              421623.9  MIRAMAR   
3              313274.8  MIRAMAR   
4              318340.5  MIRAMAR   
...                 ...      ...   
9995           365051.5  ERANGEL   
9996           364211.8  ERANGEL   
9997           388946.0  ERANGEL   
9998           351960.6  ERANGEL   
9999           625632.6  ERANGEL   

                                               match_id  time  \
0     2U4GBNA0YmnLSqvEycnTjo-KT000vfUnhSA2vfVhVPe1QB...   823   
1     2U4GBNA0YmnLSqvEycnTjo-KT000vfUnhSA2vfVhVPe1QB...   194   
2     2U4GBNA0YmnLSqvEycnTjo-KT000vfUnhSA2vfVhVPe1QB...   103   
3     2U4GBNA0YmnLSqvEycnTjo-KT000vfUnhSA2vfVhVPe1QB...  1018   
4     2U4GBNA0YmnLSqvEycnTjo-KT000vfUnhSA2vfVhVPe1QB...  1018   
...                                                 ...   ...   
9995  2U4GBNA0YmnP7JxGpV6xH481nTM-AN2Ig_D_m3DRTNjINB...  1642   
9996  2U4GBNA0YmnP7JxGpV6xH481nTM-AN2Ig_D_m3DRTNjINB...  1632   
9997  2U4GBNA0YmnP7JxGpV6xH481nTM-AN2Ig_D_m3DRTNjINB...  1608   
9998  2U4GBNA0YmnP7JxGpV6xH481nTM-AN2Ig_D_m3DRTNjINB...  1552   
9999  2U4GBNA0YmnP7JxGpV6xH481nTM-AN2Ig_D_m3DRTNjINB...   521   

          victim_name  victim_placement  victim_position_x  victim_position_y  
0     KrazyPortuguese               5.0          657725.10           146275.2  
1         X3evolution              33.0           92238.68           723375.1  
2             CtrlZee              46.0          367304.50           421216.1  
3           BlackDpre              13.0          476645.90           316758.4  
4               Vjolt              13.0          473588.50           318418.8  
...               ...               ...                ...                ...  
9995    GodlessWeasel               6.0          527811.10           375679.6  
9996        MrDrRobot               6.0          522475.80           375727.8  
9997         KrispyyK               6.0          513487.80           377484.4  
9998          Shikodd               9.0          507682.20           354356.8  
9999        Defiant1X              20.0          447384.10           625920.0  

[10000 rows x 12 columns]

**2.Change the victim names in victim_name to player1, player2,...**

In [27]:
print(kill_df['victim_name'])

0       KrazyPortuguese
1           X3evolution
2               CtrlZee
3             BlackDpre
4                 Vjolt
             ...       
9995      GodlessWeasel
9996          MrDrRobot
9997           KrispyyK
9998            Shikodd
9999          Defiant1X
Name: victim_name, Length: 10000, dtype: object


In [28]:
y=0
x=1

for _ in kill_df['victim_name']:
    kill_df.loc[y, 'victim_name'] = (f"Player {x}")
    x = x+1
    y = y+1

print(kill_df['victim_name'])

0           Player 1
1           Player 2
2           Player 3
3           Player 4
4           Player 5
            ...     
9995     Player 9996
9996     Player 9997
9997     Player 9998
9998     Player 9999
9999    Player 10000
Name: victim_name, Length: 10000, dtype: object


**3.Data Reduction (Feature selection - selecting a subset of the original features that are most relevant to the analysis):**
    <br/>
>Delete map 'MIRAMAR' while keeping map 'ERANGEL' in kill_df Display the column "map" consisting of both MIRAMAR and ERANGEL maps

In [29]:
print(kill_df['map'])

0       MIRAMAR
1       MIRAMAR
2       MIRAMAR
3       MIRAMAR
4       MIRAMAR
         ...   
9995    ERANGEL
9996    ERANGEL
9997    ERANGEL
9998    ERANGEL
9999    ERANGEL
Name: map, Length: 10000, dtype: object


In [30]:
kill_df = kill_df[kill_df.map != 'MIRAMAR']
print(kill_df['map'])

179     ERANGEL
180     ERANGEL
181     ERANGEL
182     ERANGEL
183     ERANGEL
         ...   
9995    ERANGEL
9996    ERANGEL
9997    ERANGEL
9998    ERANGEL
9999    ERANGEL
Name: map, Length: 8516, dtype: object


**4.Data cleaning (Smooth Noisy Data - removing noise from a data set, allowing important patterns to stand out):**
<br/>
>Clean the date to ie. 2017-11-20

In [31]:
print(match_df['date'])

0       2017-11-26T20:59:40+0000
1       2017-11-26T20:59:40+0000
2       2017-11-26T20:59:40+0000
3       2017-11-26T20:59:40+0000
4       2017-11-26T20:59:40+0000
                  ...           
9995    2017-11-20T01:00:32+0000
9996    2017-11-20T01:00:32+0000
9997    2017-11-20T01:00:32+0000
9998    2017-11-20T01:00:32+0000
9999    2017-11-20T01:00:32+0000
Name: date, Length: 10000, dtype: object


In [32]:
match_df['date'] = match_df['date'].astype(str).str[:10]
print(match_df['date'])

0       2017-11-26
1       2017-11-26
2       2017-11-26
3       2017-11-26
4       2017-11-26
           ...    
9995    2017-11-20
9996    2017-11-20
9997    2017-11-20
9998    2017-11-20
9999    2017-11-20
Name: date, Length: 10000, dtype: object


**5.Data Transformation (Data normalisation - standardizing the data to a common scale or range):**
<br/>
>Convert 'player_survive_time' and 'time' in seconds to time in minutes, ie. 75 seconds = 1.25 min

In [33]:
print(match_df['player_survive_time'])

0       1106.320
1       1106.315
2        235.558
3        197.553
4       1537.495
          ...   
9995    1331.496
9996    1161.466
9997     361.651
9998     497.617
9999     495.147
Name: player_survive_time, Length: 10000, dtype: float64


In [34]:
match_df['player_survive_time'] = match_df['player_survive_time'] /60
print(match_df['player_survive_time'])

0       18.438667
1       18.438583
2        3.925967
3        3.292550
4       25.624917
          ...    
9995    22.191600
9996    19.357767
9997     6.027517
9998     8.293617
9999     8.252450
Name: player_survive_time, Length: 10000, dtype: float64


In [35]:
print(kill_df['time'])

179     1035
180     1035
181     1422
182     1210
183     1818
        ... 
9995    1642
9996    1632
9997    1608
9998    1552
9999     521
Name: time, Length: 8516, dtype: int64


In [36]:
kill_df['time'] = kill_df['time'] /60
print(kill_df['time'])


179     17.250000
180     17.250000
181     23.700000
182     20.166667
183     30.300000
          ...    
9995    27.366667
9996    27.200000
9997    26.800000
9998    25.866667
9999     8.683333
Name: time, Length: 8516, dtype: float64


C:\Users\Justin\AppData\Local\Temp\ipykernel_15512\13429439.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kill_df['time'] = kill_df['time'] /60


**6.Data cleaning (Checking missing values):**
<br/>
>Show duplictes, NaN and empty values and clean the values
<br/>

Check for duplicated values in match_df and kill_df

In [37]:
match_df.duplicated().sum()

0

In [38]:
kill_df.duplicated().sum()

0

--> Since there are no duclipated values, we do not need to delete them
<br/>
<br/>
<br/>
Check Null values inside match_df

In [39]:
match_df.isnull().sum()

date                    0
game_size               0
match_id                0
match_mode              0
party_size              0
player_assists          0
player_dbno             0
player_dist_ride        0
player_dist_walk        0
player_dmg              0
player_kills            0
player_name            13
player_survive_time     0
team_id                 0
team_placement          0
dtype: int64

--> There are 13 Null values at column "Player_name"

Check Null values inside kill_df

In [40]:
kill_df.isnull().sum()

killed_by              0
killer_name          611
killer_placement     611
killer_position_x    611
killer_position_y    611
map                   97
match_id               0
time                   0
victim_name            0
victim_placement     186
victim_position_x      0
victim_position_y      0
dtype: int64

--> There are 611 Null values at column

>"killer_name"
<br/>
>"killer_placement"
<br/>
>"killer_position_x"
<br/>
>"killer_position_y"


--> There are 97 Null values and 186 Null values at columns

>"map"
<br/>
>"victim_placement"

**7.Data cleaning (Handling missing values - Replacing missing values with a constant):**
<br/>
>Change the null values at column "player_name" to “unknown”

In [41]:
match_df['player_name'].fillna('unknown', inplace=True)
match_df.isnull().sum()

date                   0
game_size              0
match_id               0
match_mode             0
party_size             0
player_assists         0
player_dbno            0
player_dist_ride       0
player_dist_walk       0
player_dmg             0
player_kills           0
player_name            0
player_survive_time    0
team_id                0
team_placement         0
dtype: int64

Display rows that consist Null value killer_name

In [42]:
null_rows = kill_df[kill_df["killer_name"].isna()]
(null_rows)

killed_by killer_name  killer_placement  killer_position_x  \
187   Down and Out         NaN               NaN                NaN   
204       Bluezone         NaN               NaN                NaN   
209       Bluezone         NaN               NaN                NaN   
223       Bluezone         NaN               NaN                NaN   
229       Bluezone         NaN               NaN                NaN   
...            ...         ...               ...                ...   
9940      Bluezone         NaN               NaN                NaN   
9955         Drown         NaN               NaN                NaN   
9957         Drown         NaN               NaN                NaN   
9958      Bluezone         NaN               NaN                NaN   
9970  Down and Out         NaN               NaN                NaN   

      killer_position_y      map  \
187                 NaN  ERANGEL   
204                 NaN  ERANGEL   
209                 NaN  ERANGEL   
223                 NaN  ERANGEL   
229                 NaN  ERANGEL   
...                 ...      ...   
9940                NaN  ERANGEL   
9955                NaN  ERANGEL   
9957                NaN  ERANGEL   
9958                NaN  ERANGEL   
9970                NaN  ERANGEL   

                                               match_id       time  \
187   2U4GBNA0YmnRe95wOy7kuweIkuZo5Roa0WjNZsgboi2gzz...  27.433333   
204   2U4GBNA0YmnRe95wOy7kuweIkuZo5Roa0WjNZsgboi2gzz...  27.433333   
209   2U4GBNA0YmnRe95wOy7kuweIkuZo5Roa0WjNZsgboi2gzz...  29.800000   
223   2U4GBNA0YmnRe95wOy7kuweIkuZo5Roa0WjNZsgboi2gzz...  31.116667   
229   2U4GBNA0YmnRe95wOy7kuweIkuZo5Roa0WjNZsgboi2gzz...  22.716667   
...                                                 ...        ...   
9940  2U4GBNA0Yml3Tos6nrlUNhrFTSpEFwANNAnIUrRGhVYsH7...  31.466667   
9955  2U4GBNA0Yml3Tos6nrlUNhrFTSpEFwANNAnIUrRGhVYsH7...  16.250000   
9957  2U4GBNA0Yml3Tos6nrlUNhrFTSpEFwANNAnIUrRGhVYsH7...   4.300000   
9958  2U4GBNA0Yml3Tos6nrlUNhrFTSpEFwANNAnIUrRGhVYsH7...  18.333333   
9970  2U4GBNA0Yml3Tos6nrlUNhrFTSpEFwANNAnIUrRGhVYsH7...  23.583333   

      victim_name  victim_placement  victim_position_x  victim_position_y  
187    Player 188               8.0           475198.1           426731.2  
204    Player 205               8.0           468805.0           430984.8  
209    Player 210               4.0           498605.2           442824.2  
223    Player 224               2.0           503327.2           426248.7  
229    Player 230              40.0                0.0                0.0  
...           ...               ...                ...                ...  
9940  Player 9941               2.0           128437.4           388790.9  
9955  Player 9956              23.0           284318.3           276540.6  
9957  Player 9958              37.0           342167.7           532223.9  
9958  Player 9959              22.0           366638.9           411980.6  
9970  Player 9971              13.0           200591.5           383582.4  

[611 rows x 12 columns]

Display the row shape consist null value killer_name

In [43]:
null_rows.shape

(611, 12)

--> we know that 611 of the null value killer_name
is caused by bluezone, drown, naz, drop and down which is not by a real human enemy. Therefore the killer_name is presented as null
<br/>
<br/>

Replace Null values at column "killer_name" to unknown

In [44]:
kill_df['killer_name'].fillna('unknown', inplace=True)

C:\Users\Justin\AppData\Local\Temp\ipykernel_15512\775270850.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kill_df['killer_name'].fillna('unknown', inplace=True)


In [45]:
kill_df.isnull().sum()

killed_by              0
killer_name            0
killer_placement     611
killer_position_x    611
killer_position_y    611
map                   97
match_id               0
time                   0
victim_name            0
victim_placement     186
victim_position_x      0
victim_position_y      0
dtype: int64

Delete row that consist null values in victim_placement

In [46]:
kill_df.dropna(subset = ["victim_placement"] , inplace=True)
kill_df.dropna(subset = ["map"] , inplace=True)
kill_df.isnull().sum()

C:\Users\Justin\AppData\Local\Temp\ipykernel_15512\2244949287.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kill_df.dropna(subset = ["victim_placement"] , inplace=True)
C:\Users\Justin\AppData\Local\Temp\ipykernel_15512\2244949287.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kill_df.dropna(subset = ["map"] , inplace=True)


killed_by              0
killer_name            0
killer_placement     604
killer_position_x    604
killer_position_y    604
map                    0
match_id               0
time                   0
victim_name            0
victim_placement       0
victim_position_x      0
victim_position_y      0
dtype: int64

--> Only the killer_name with null values is cleaned
This is because the killer_placement, killer_position x and y are qualitative data and thus cannot be measured into a value


In [47]:
match_df.isnull().sum()

date                   0
game_size              0
match_id               0
match_mode             0
party_size             0
player_assists         0
player_dbno            0
player_dist_ride       0
player_dist_walk       0
player_dmg             0
player_kills           0
player_name            0
player_survive_time    0
team_id                0
team_placement         0
dtype: int64

**8.Dsiplay the final version of our cleaned datasets**

In [50]:
match_df

date  game_size  \
0     2017-11-26         37   
1     2017-11-26         37   
2     2017-11-26         37   
3     2017-11-26         37   
4     2017-11-26         37   
...          ...        ...   
9995  2017-11-20         29   
9996  2017-11-20         29   
9997  2017-11-20         29   
9998  2017-11-20         29   
9999  2017-11-20         29   

                                               match_id match_mode  \
0     2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
1     2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
2     2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
3     2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
4     2U4GBNA0YmnNZYkzjkfgN4ev-hXSrak_BSey_YEG6kIuDG...        tpp   
...                                                 ...        ...   
9995  2U4GBNA0Ymlm4JHKbbmQ9x9_rZbdELcOlVMjnXNbpV6MTm...        tpp   
9996  2U4GBNA0Ymlm4JHKbbmQ9x9_rZbdELcOlVMjnXNbpV6MTm...        tpp   
9997  2U4GBNA0Ymlm4JHKbbmQ9x9_rZbdELcOlVMjnXNbpV6MTm...        tpp   
9998  2U4GBNA0Ymlm4JHKbbmQ9x9_rZbdELcOlVMjnXNbpV6MTm...        tpp   
9999  2U4GBNA0Ymlm4JHKbbmQ9x9_rZbdELcOlVMjnXNbpV6MTm...        tpp   

      party_size  player_assists  player_dbno  player_dist_ride  \
0              2               0            1        2870.72400   
1              2               0            1        2938.40723   
2              2               0            0           0.00000   
3              2               0            0           0.00000   
4              2               0            0        2619.07739   
...          ...             ...          ...               ...   
9995           4               0            2        5947.79000   
9996           4               0            0        4421.93262   
9997           4               0            2           0.00000   
9998           4               0            1           0.00000   
9999           4               1            0           0.00000   

      player_dist_walk  player_dmg  player_kills    player_name  \
0          1784.847780         117             1       SnuffIes   
1          1756.079710         127             1         Ozon3r   
2           224.157562          67             0         bovize   
3            92.935150           0             0        sbahn87   
4          2510.447000         175             2      GeminiZZZ   
...                ...         ...           ...            ...   
9995       1629.568000          87             2       2thikk4u   
9996       3212.797850         137             2   peckerpecker   
9997        873.056300         193             0      Bawngfist   
9998        744.918152          87             1  Coots_McGoots   
9999        670.910339          12             0          sbwwt   

      player_survive_time  team_id  team_placement  
0               18.438667        4              18  
1               18.438583        4              18  
2                3.925967        5              33  
3                3.292550        5              33  
4               25.624917       14              11  
...                   ...      ...             ...  
9995            22.191600        6               7  
9996            19.357767        6               7  
9997             6.027517        6               7  
9998             8.293617        7              14  
9999             8.252450        7              14  

[10000 rows x 15 columns]

In [51]:
kill_df

killed_by      killer_name  killer_placement  killer_position_x  \
179   Down and Out        Malcolm_x               9.0           496989.8   
180          M16A4        Malcolm_x               9.0           496989.8   
181            AKM           G_Berg               7.0           460416.7   
182            AKM      Lukesnake17              20.0           488034.1   
183            SKS         AlooGobi               2.0           501062.9   
...            ...              ...               ...                ...   
9995           SKS  dragonfruitbamf               2.0           536215.7   
9996          M416           lddoos               3.0           550700.4   
9997  Down and Out            Nan_P               3.0           507691.3   
9998       Grenade         brentech               7.0           503235.2   
9999        SCAR-L             SwwH              13.0           447508.2   

      killer_position_y      map  \
179            312569.7  ERANGEL   
180            312569.7  ERANGEL   
181            414748.8  ERANGEL   
182            347220.3  ERANGEL   
183            425078.6  ERANGEL   
...                 ...      ...   
9995           365051.5  ERANGEL   
9996           364211.8  ERANGEL   
9997           388946.0  ERANGEL   
9998           351960.6  ERANGEL   
9999           625632.6  ERANGEL   

                                               match_id       time  \
179   2U4GBNA0YmnRe95wOy7kuweIkuZo5Roa0WjNZsgboi2gzz...  17.250000   
180   2U4GBNA0YmnRe95wOy7kuweIkuZo5Roa0WjNZsgboi2gzz...  17.250000   
181   2U4GBNA0YmnRe95wOy7kuweIkuZo5Roa0WjNZsgboi2gzz...  23.700000   
182   2U4GBNA0YmnRe95wOy7kuweIkuZo5Roa0WjNZsgboi2gzz...  20.166667   
183   2U4GBNA0YmnRe95wOy7kuweIkuZo5Roa0WjNZsgboi2gzz...  30.300000   
...                                                 ...        ...   
9995  2U4GBNA0YmnP7JxGpV6xH481nTM-AN2Ig_D_m3DRTNjINB...  27.366667   
9996  2U4GBNA0YmnP7JxGpV6xH481nTM-AN2Ig_D_m3DRTNjINB...  27.200000   
9997  2U4GBNA0YmnP7JxGpV6xH481nTM-AN2Ig_D_m3DRTNjINB...  26.800000   
9998  2U4GBNA0YmnP7JxGpV6xH481nTM-AN2Ig_D_m3DRTNjINB...  25.866667   
9999  2U4GBNA0YmnP7JxGpV6xH481nTM-AN2Ig_D_m3DRTNjINB...   8.683333   

       victim_name  victim_placement  victim_position_x  victim_position_y  
179     Player 180              22.0           497385.4           331528.2  
180     Player 181              22.0           497819.4           331981.3  
181     Player 182              16.0           459817.9           414426.3  
182     Player 183               9.0           487444.2           347651.0  
183     Player 184               3.0           493043.4           434458.1  
...            ...               ...                ...                ...  
9995   Player 9996               6.0           527811.1           375679.6  
9996   Player 9997               6.0           522475.8           375727.8  
9997   Player 9998               6.0           513487.8           377484.4  
9998   Player 9999               9.0           507682.2           354356.8  
9999  Player 10000              20.0           447384.1           625920.0  

[8234 rows x 12 columns]

**9.Save the dataset to new csv as finished version for later use**

In [52]:
match_df.to_csv('final_agg_match_stats.csv', index=False)

In [53]:
kill_df.to_csv('final_kill_match_stats.csv', index=False)